crates/mun_memory/src/diff.rs

```diff
@@ -2,25 +2,29 @@ pub mod myers;
 
```

Comment:
The code in this file is pretty complex. I think it could definitely use more comments to explain what is going on. Not just what the code is doing exactly but more the entire process. I still find the code quite daunting.